packages/pg-connection-string/index.js

```diff
@@ -49,7 +49,7 @@ function parse(str) {
   if (!config.host) {
     // Only set the host if there is no equivalent query param.
     config.host = decodeURIComponent(hostname)
-  } else if (hostname) {
+  } else if (hostname && /^%2f/i.test(hostname)) {
```

Comment:
Nice fix. Should probably add a comment here explaining the logic.

Eg.

```suggestion
  } else if (hostname && /^%2f/i.test(hostname)) {
    // Only prepend the hostname to the pathname if it is not a URL encoded Unix socket host.
```